katdal/applycal.py

```diff
@@ -38,6 +42,106 @@
 logger = logging.getLogger(__name__)
 
 
+def _call_from_block_function(func, shape, num_chunks, chunk_location, array_location, func_kwargs):
```

Comment:
I didn't want to do that (in dask itself) because then it's not extensible: if you then want to pass an extra parameter, you'll get an error if the function isn't expecting it. There is some discussion happening in dask/dask#4476 about the API though, so once that's settled I'll transfer whatever we agree to here (and eventually we'll just delete this code and use the code in dask :-).

Reply:
That's a much better reason (and it has happened to `da.map_blocks` before when the keys of `block_info` changed). Happy with that.